### PR TITLE
docs: Add note on resource-led policy design in `Best practices` section

### PR DESCRIPTION
**Pull request by Sambigeara**
Signed-off-by: Sam Lock <sam@swlock.co.uk>

### CODE REVIEW

**Review by charithe — COMMENTED** [review]
I am struggling a bit to understand this new section. My understanding was that this page is about how to approach writing the rules for a resource, which you can do by thinking along the dimensions of actions, roles and so on. I don't think it can be resource-led because  that's a given (we are writing a resource policy, after all).

Would it be better to add an intro section to the document to say the following.
- With Cerbos, access rules are always resource oriented. You must write policies targeting each resource you want to secure.
- Within a policy, there are several approaches you can use to write access rules (action-led, role-led etc.)

**Comment by Sambigeara** [comment]
Yeah, it certainly didn't seem _quite_ right putting it here. The intention was to solve [this user request](https://cerbos.slack.com/archives/C03JN3JM2LC/p1673947852199509), but perhaps this would be better placed elsewhere. We're currently waiting on feedback from the user in the community Slack - I'll wait for that and then consider better alternatives. I like your suggestion on some preliminary notes in an intro.

File: docs/modules/policies/pages/best_practices.adoc [hunk 1]
```diff
@@ -119,6 +119,37 @@ Finally, we can opt to explicitly state which actions a user *cannot* do:
 
 This would suit scenarios where a principal can perform _nearly_ every action, and you want to explicitly list disallowed actions.
 
+=== Attribute-led
+
+Consider the following hypothetical scenario:
+
+****
+An organization models it's resources as specific _data sets_. Each data set is unique, as are the principals trying to access them. The organization uses JWTs extensively to manage and transmit identity/contextual information. The resource policies map 1:1 to each data set, and access is governed by arbitrary information (in this case, passed within the JWT).
```
Comment (charithe) [inline docs/modules/policies/pages/best_practices.adoc:127]:
```suggestion
An organization models its resources as specific _data sets_. Each data set is unique, as are the principals trying to access them. The organization uses JWTs extensively to manage and transmit identity/contextual information. The resource policies map 1:1 to each data set, and access is governed by arbitrary information (in this case, passed within the JWT).
```

File: docs/modules/policies/pages/best_practices.adoc [hunk 2]
```diff
@@ -119,6 +119,37 @@ Finally, we can opt to explicitly state which actions a user *cannot* do:
...
+****
+
+Given the changeable nature of the accessing audiences, it doesn't necessarily make sense to rely on roles for access. What we could do instead is to globally allow all roles and actions and then determine access based on attributes passed in the JWT. Take a look at the following example policy:
```
Comment (charithe) [inline docs/modules/policies/pages/best_practices.adoc:130]:
```suggestion
Given the dynamic nature of audiences, it's not practical to enumerate all roles that have access. What we could do instead is to globally allow all roles and actions and then determine access based on attributes passed in the JWT. Take a look at the following example policy:
```

**Pull request merged**